Dockerfile

```diff
@@ -1,35 +1,83 @@
-FROM --platform=$BUILDPLATFORM golang:1.18-alpine as BUILD
+FROM --platform=$BUILDPLATFORM golang:1.19-alpine3.16 AS build-env
 
-WORKDIR /relayer
+RUN apk add --update --no-cache curl make git libc-dev bash gcc linux-headers eudev-dev
 
-# Update and install needed deps prioir to installing the binary.
-RUN apk update && \
-  apk --no-cache add make git build-base
+ARG TARGETARCH
+ARG BUILDARCH
 
-# Copy go.mod and go.sum first and download for caching go modules
-COPY go.mod go.sum ./
+RUN if [ "${TARGETARCH}" = "arm64" ] && [ "${BUILDARCH}" != "arm64" ]; then \
+        wget -c https://musl.cc/aarch64-linux-musl-cross.tgz -O - | tar -xzvv --strip-components 1 -C /usr; \
+    elif [ "${TARGETARCH}" = "amd64" ] && [ "${BUILDARCH}" != "amd64" ]; then \
+        wget -c https://musl.cc/x86_64-linux-musl-cross.tgz -O - | tar -xzvv --strip-components 1 -C /usr; \
+    fi
 
-RUN go mod download
+ADD . .
 
-# Copy the files from host
-COPY . .
+RUN set -eux; \
+    LIBDIR=/lib; \
+    if [ "${TARGETARCH}" = "arm64" ]; then \
+      ARCH=aarch64; \
+      if [ "${BUILDARCH}" != "arm64" ]; then \
+        LIBDIR=/usr/aarch64-linux-musl/lib; \
+        mkdir -p $LIBDIR; \
+        export CC=aarch64-linux-musl-gcc CXX=aarch64-linux-musl-g++;\
+      fi;\
+    elif [ "${TARGETARCH}" = "amd64" ]; then \
+      ARCH=x86_64; \
+      if [ "${BUILDARCH}" != "amd64" ]; then \
+        LIBDIR=/usr/x86_64-linux-musl/lib; \
+        mkdir -p $LIBDIR; \
+        export CC=x86_64-linux-musl-gcc CXX=x86_64-linux-musl-g++; \
+      fi; \
+    fi;
 
-ARG TARGETARCH TARGETOS
-ENV GOOS=${TARGETOS} GOARCH=${TARGETARCH}
-RUN make install
+RUN GOOS=linux GOARCH=$TARGETARCH CGO_ENABLED=1 LDFLAGS='-linkmode external -extldflags "-static"' make install
```

Comment:
Are CC and CXX vars needed for the make install? if so, this should be grouped with L16-L32. Looks like we don't need ARCH or the mkdir -p $LIBDIR. If we don't need CC and CXX vars for the make install, then we can remove L16-L32

Reply:
We almost certainly need the setting of the C/C++ compiler...